diff --git a/src/GUI/FGCocoaMenuBar.mm b/src/GUI/FGCocoaMenuBar.mm
index f34fc14..e545839 100644
--- a/src/GUI/FGCocoaMenuBar.mm
+++ b/src/GUI/FGCocoaMenuBar.mm
@@ -39,6 +39,18 @@ public:
   MenuItemBindings itemBindings;
 };
 
+#if __MAC_OS_X_VERSION_MIN_REQUIRED < 1060
+@protocol NSApplicationDelegate <NSObject>
+@end
+@protocol NSWindowDelegate <NSObject>
+@end
+@protocol NSTableViewDataSource <NSObject>
+@end
+@protocol NSTableViewDelegate <NSObject>
+@end
+@protocol NSMenuDelegate <NSObject>
+@end
+#endif
 
 @interface CocoaMenuDelegate : NSObject <NSMenuDelegate> {
 @private
