--- simgear/scene/tsync/terrasync.cxx	2013-02-12 00:17:50.000000000 +0100
+++ simgear/scene/tsync/terrasync_105.cxx	2013-02-12 00:19:19.000000000 +0100
@@ -399,18 +399,7 @@
     apr_pool_t *subpool = svn_pool_create(_svn_pool);
 
     svn_error_t *err = NULL;
-#if (SVN_VER_MINOR >= 5)
-    err = svn_client_checkout3(NULL,
-            command.str().c_str(),
-            dest_base_dir.str().c_str(),
-            _svn_rev_peg,
-            _svn_rev,
-            svn_depth_infinity,
-            0, // ignore-externals = false
-            0, // allow unver obstructions = false
-            _svn_ctx,
-            subpool);
-#else
+
     // version 1.4 API
     err = svn_client_checkout2(NULL,
             command.str().c_str(),
@@ -421,7 +410,6 @@
             0, // ignore externals = false
             _svn_ctx,
             subpool);
-#endif
 
     bool ReturnValue = true;
     if (err)
@@ -635,15 +623,9 @@
 
     svn_auth_baton_t *ab=NULL;
 
-#if (SVN_VER_MINOR >= 6)
-    err = svn_cmdline_create_auth_baton  (&ab,
-            TRUE, NULL, NULL, config_dir, TRUE, FALSE, cfg,
-            _svn_ctx->cancel_func, _svn_ctx->cancel_baton, pool);
-#else
     err = svn_cmdline_setup_auth_baton(&ab,
         TRUE, NULL, NULL, config_dir, TRUE, cfg,
         _svn_ctx->cancel_func, _svn_ctx->cancel_baton, pool);
-#endif
 
     if (err)
         return svn_cmdline_handle_exit_error(err, pool, "fgfs: ");
